gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -761,6 +735,22 @@ def extract_options(kwargs)
         }
       end
 
+      # allows using either a Hash or Request object
+      # Handles either url or uri
+      def extract_request(request)
+        if request.is_a?(Hash)
+          request[:url] ||= request[:uri]
+          return request
+        end
+
+        {
+          http_method: request.http_method,
+          url: request.respond_to?(:url) ? request.url : request.uri,
+          headers: request.headers.to_h,
```

Comment:
We should handle a headers object similarly - I believe request.headers should already support enumeration based on how it's used in here? Can ditch to_h

Reply:
We get errors:
```
NoMethodError:
         undefined method `transform_keys' for an instance of Hearth::HTTP::Fields::Proxy
```

So I think it makes sense to use to_h here (and doesn't ever hurt if its already a hash).